sklearn/tests/test_pipeline.py

```diff
@@ -1679,3 +1680,107 @@ def test_feature_union_feature_names_in_():
     union = FeatureUnion([("pass", "passthrough")])
     union.fit(X_array)
     assert not hasattr(union, "feature_names_in_")
+
+
+# Test that metadata is routed correctly for pipelines
+# ====================================================
+
+
+class SimpleEstimator(BaseEstimator):
+    # This class is used in this section for testing routing in the pipeline.
+    # This class should have every set_{method}_request
```

Comment:
In these set of methods we are verifying that sample_weight is received correctly. But I think we are also passing prop="a" in the test below. Would it make sense to also verify that prop is received for any methods that are expected to receive it?